client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
      * 
      * @param stream Stream object.
      * @param sealedSegments List of segments to be sealed.
      * @param newKeyRanges Key ranges after scaling the stream.
+     * @param timeout timeout and throw exception if scale does not complete withing specified timeout.
+     * @param executorService executor to be used for busy waiting.
      * @return A future which will throw if the operation fails, otherwise returning a boolean to
-     *         indicate that the stream was scaled or false if the segments to seal have already
-     *         been sealed.
+     *         indicate that the scaling was started or not.
      */
     CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
```

Comment:
i see where you are coming from and it makes sense.. 
here are my two cents on it: the interface guarantees the client that once called, this method, scalestream will return after either having successfully completed the scale (succeeded or failed) or timed out.
until then the future returned by it, if you do a get on it, will block.

Now it is our implementation detail that instead of providing a blocking api on controller service with corresponding rpc call, we want to implement it using the start scale and check scale status apis provided by controller. Does that make sense?